src/lib/createTree.js

```diff
@@ -40,19 +39,28 @@ const createTree = (treeStructure, nodesWebhooks) => {
         contributionWebhook,
         aggregationWebhook,
         level: depth,
+        role:
+          depth === 0
+            ? 'Querier'
+            : depth === treeStructure.depth - 2
+            ? 'Leaf'
+            : depth === treeStructure.depth - 1
+            ? 'Contributor'
+            : 'Aggregator',
```

Comment:
It's a matter of style and it's not blocking, but I really dislike nested ternaries, I find quite hard to read and I put much more value in readability rather than code elegance.